### PR TITLE
tests: cleanup

### PR DESCRIPTION
**Pull request by ignasi35**
Soem cleanup after merging #3581 and a few other competing PRs

### CODE REVIEW

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2FrameProbe.scala [hunk 1]
```diff
@@ -41,15 +41,16 @@ private[http2] trait Http2FrameProbe {
...
-  def expectSettingsAck(): Unit
-
   def expectFrame(frameType: FrameType, expectedFlags: ByteFlag, streamId: Int, payload: ByteString): Unit
 
   def expectFramePayload(frameType: FrameType, expectedFlags: ByteFlag, streamId: Int): ByteString
   def expectFrameFlagsAndPayload(frameType: FrameType, streamId: Int): (ByteFlag, ByteString)
   def expectFrameFlagsStreamIdAndPayload(frameType: FrameType): (ByteFlag, Int, ByteString)
 
   def expectSETTINGS(): SettingsFrame
+  def expectSettingsAck(): Unit
```
Comment (raboof) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2FrameProbe.scala:51]: :+1:

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 2]
```diff
@@ -995,25 +995,30 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
...
         val request = HttpRequest(uri = "https://www.example.com/")
+
         // client set SETTINGS_MAX_CONCURRENT_STREAMS to 1 so an attempt from the server to open more streams
         // should fail. But as long as the outgoing streams are a result of client-initiated communication
         // they should succeed.
-        sendHEADERS(1, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
-        sendHEADERS(3, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
-        sendHEADERS(5, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
-        sendHEADERS(7, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+        openStream(1)
```
Comment (raboof) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:1006]: I kindof liked the explicit sending of request frames and response objects, makes it easier to see who is triggering the opening/closing. I'm OK with this as well, though.

**Review by raboof — APPROVED** [review]

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 3]
```diff
@@ -995,25 +995,30 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       "not limit response streams even when the client send a SETTINGS_MAX_CONCURRENT_STREAMS" in new TestSetup(
         Setting(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 1)
       ) with RequestResponseProbes {
+        def openStream(streamId: Int) = sendHEADERS(streamId, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
```
Comment (jlprat) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:998]: As you are using visual break lines further down, I would also add a break line here to create some visual space between the `def` and the previous line.

**Review by jlprat — APPROVED** [review]
LGTM only one minor cosmetic comment

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala (hunk 2 above)
Comment (jrudolph) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:1006]: I agree with @raboof, but I think it's fine in the context of a single test.

**Review by jrudolph — APPROVED** [review]
LGTM, thanks.

**Pull request merged**